pkg/controllers/node/termination/controller.go

```diff
@@ -138,6 +152,56 @@ func (c *Controller) removeFinalizer(ctx context.Context, n *v1.Node) error {
 	return nil
 }
 
+func (c *Controller) terminationGracePeriodExpirationTime(ctx context.Context, node *v1.Node) (*time.Time, error) {
+
+	nodeClaim := &v1beta1.NodeClaim{}
+
+	if len(node.OwnerReferences) == 0 {
+		logging.FromContext(ctx).Errorf("node has no owner, could not find NodeClaim for Node: %v", node.Name)
+		return nil, nil
+	}
+
+	nodeClaimName := types.NamespacedName{}
+	for _, o := range node.OwnerReferences {
+		// verify that we're finding Karpenter's NodeClaim of any version permutation
+		if strings.HasPrefix(o.APIVersion, "karpenter.sh") && o.Kind == "NodeClaim" {
```

Comment:
Should we use `schema.ParseGroupVersion()` to make a more rigorous check here against the group name?